tangelo/toolboxes/post_processing/post_selection.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2022 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module provides functions to create symmetry verification and postselection circuits."""
+from tangelo.linq import Circuit, Gate
+from tangelo.linq.helpers import measurement_basis_gates, pauli_string_to_of
+
+
+def ancilla_symmetry_circuit(circuit, sym_op):
+    """Appends to a circuit a symmetry operator and ancilla for symmetry verification.
+    For more details see arXiv:1807.10050.
+
+    Args:
+        circuit (Circuit): A quantum circuit to be equipped with symmetry verification
+        sym_op (string): The symmetry operator to be verified, in the form of a Pauli string
+
+    Returns:
+        Circuit: The input circuit appended with the proper basis rotation
+            and entanglement with an ancilla qubit"""
+    n_qubits = circuit.width
+    basis_gates = measurement_basis_gates(pauli_string_to_of(sym_op))
+    basis_circ = Circuit(basis_gates)
+    parity_gates = [Gate("CNOT", n_qubits, i) for i in range(n_qubits)]
+    circuit_new = circuit + basis_circ + Circuit(parity_gates) + basis_circ.inverse()
+    return circuit_new
+
+
+def post_select(hist, expected_outcomes):
+    """Applies post selection to frequency data based on
```

Comment:
The custom is not to conjugate the verbs in docstrings and comments, when you start with a verb like that.
`Apply post-selection to....`